src/open_petro_elastic/config/fluids.py

```diff
@@ -49,14 +50,17 @@ class Fluids:
     )
     mix_method: Literal["wood", "brie"] = "wood"
     brie_constant: float = 3.0
-    index_of_gas: int = 0
+    index_of_gas: List[int] = field(default_factory=list)
     fluid_model: Literal[tuple(fluid_model_providers.keys())] = "default"
 
     def __post_init_post_parse__(self):
         fix_one_fraction(self.constituents)
-        gas_constituents = [c for c in self.constituents if c.material.type == "gas"]
-        if len(gas_constituents) > 0:
-            self.index_of_gas = self.constituents.index(gas_constituents[0])
+        if len(self.index_of_gas) == 0:
+            self.index_of_gas = [
+                i
+                for i, c in enumerate(self.constituents)
+                if c.material.type in ("gas", "carbon_dioxide")
```

Comment:
I think maybe introducing a separate constituent list in the Fluids class for gasses would be more clean:
```
fluids:
    mix_method: "wood" # either brie or wood
    temperature: 75.0 # celsius
    gasses:
            - material:
            type: "carbon_dioxide"
            interpolate_density: true
          fraction: # Ratio of gas to remaining fluid
              column: "gas_saturation"
          fluid_model: "span_wagner"
    constituents:
        - material:
            type: "brine"
            salinity: 45000 # ppm
          fraction: # Ratio of brine to remaining fluid
              column: "brine_saturation"
```

Reply:
Getting rid `index_of_gas` would definitely make this clearer. Using a separate "gases" keyword is better, but I agree with Jimmy that it might be a bit ambiguous. Since its only relevant for Brie mixing it might be confusing since it appears to have a greater significance than it might have (ie no significance for Wood). There is also the terminology aspect: CO2 is likely to be super-critical and not vapor, making the "gas" term a potential source of confusion.

We could view this as a setting that is related specifically to Brie mixing in the same way as the `brie_constant` parameter. Could it perhaps be treated in a similar manner? E.g. as a separate keyword under fluids ("Option 1") that determines which material types that are treated as a "Brie-gas", or as a flag per constituent ("Option 2"):

```
fluids:
    mix_method: "brie" # either brie or wood
    temperature: 75.0 # celsius
    brie_constant: 5
    brie_gases:  # Option 1
        - "carbon_dioxide"
        - "gas"
    constituents:
        - material:
            type: "carbon_dioxide"
            interpolate_density: true
          fraction: # Ratio of gas to remaining fluid
              column: "gas_saturation"
          fluid_model: "span_wagner"
          gaseous: true  # Option 2
        - material:
            type: "brine"
            salinity: 45000 # ppm
          fraction: # Ratio of brine to remaining fluid
              column: "brine_saturation"
```

Reply:
I think option 1 might be problematic as constituents do not really have a "name", just a type. I think Option 2 will be better, but implementation-wise cause some entanglement between brie and constituents.

I think maybe a combination where you add a "name" field to constituents and a "brie_gases" list which lists the names of constituents you want to be considered gas in the brie calculation?

Reply:
well, I also prefers something that look like option 1 over option 2, as it is strictly related to brie. But I see the problem @eivindjahren raises. 2 possibilities there in my opinion:
- either we consider there is only 1 constituent of each fluid type and then we can use them as identifiers
- either we can attribute id fields for each fluid constituent and refer to it in the `brie_gases` option (if not provided, the id could be by default `"fluid_constituent_<i>"` with i being the i-th constituent of the list, or something like that)

Reply:
The simplest solution would be for "brie_gases" to hold indices, which I think is still better than "index_of_gas" 
```yaml
    brie_gases: 
        - 0
        - 1
    constituents:
        - material:
            type: "carbon_dioxide"
            interpolate_density: true
          fraction: # Ratio of gas to remaining fluid
              column: "gas_saturation"
          fluid_model: "span_wagner" 
        - material:
            type: "gas"
        - material:
            type: "brine"
            salinity: 45000 # ppm
          fraction: # Ratio of brine to remaining fluid
              column: "brine_saturation"
```